### PR TITLE
add plugin for language switcher

### PR DESCRIPTION
**Pull request by jsjexpert**
## Description

Added a language switcher plugin for NextJS i18n projects to quickly swap between locales within the same content item.

_Screenshot_

### CODE REVIEW

File: plugins/localization/package.json [hunk 1]
```diff
@@ -0,0 +1,123 @@
+{
+  "name": "@wecre8websites/builder-localization",
```
Comment (teleaziz) [inline plugins/localization/package.json:2]: `@wecre8websites/builder-localization `, this should start with `@builder.io` so we can release it to npm, also I think this extension is more for updating preview URL per locale, what do you think about naming it something closer to that, something like `Localized preview plugin`

Comment (teleaziz) [inline plugins/localization/package.json:2]:
```suggestion
  "name": "@builder.io/plugin-localized-preview",
```

File: plugins/localization/src/components/language-switcher.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        onChange={v => {
+          if (
+            context.designerState.editingIframeRef &&
+            context.designerState.editingContentModel?.url
+          ) {
+            const { origin, search } = new URL(
+              context.designerState.editingIframeRef.src
+            )
+            if (!!v.target.dataset.default && v.target.value === v.target.dataset.default) {
+              context.designerState.editingIframeRef.src =
```
Comment (teleaziz) [inline plugins/localization/src/components/language-switcher.tsx:67]: have you tried: `context.designerState.editingContentModel.previewUrl = `

File: plugins/localization/src/plugin.ts [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+interface OnSaveActions {
+  updateSettings(partal: Record<string, any>): Promise<void>;
+  addModel(model: Model): Promise<void>;
+}
+
+
+
+Builder.register('plugin', {
+  id: pluginId,
+  name: 'wecre8Websites Localization',
```
Comment (teleaziz) [inline plugins/localization/src/plugin.ts:56]:
```suggestion
  name: 'Localized Preview',
```

File: plugins/localization/src/plugin.ts [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+Builder.register('app.onLoad', async ({ triggerSettingsDialog }: AppActions) => {
+  const currentOrg = context.user.organization;
+  const pluginSettings = currentOrg.value.settings.plugins.get(pluginId);
+  const locales = pluginSettings?.get('locales');
+  if (typeof locales === "string" || !locales) {
+    pluginSettings.set("locales", []);
+    await triggerSettingsDialog(pluginId);
+  }
+  window.languageSettingsTrigger = async () => await triggerSettingsDialog(pluginId)
+});
```
Comment (teleaziz) [inline plugins/localization/src/plugin.ts:130]: end of file

File: plugins/localization/src/components/language-switcher.tsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+  >()
+  const onLoad = async () => {
+    const currentUser = await context.user.getUser(context.user.id)
+    //@ts-ignore
+    const currentOrg = await context.user.organization
+    //@ts-ignore
+    const pluginSettings = await (await currentOrg.value.settings.plugins).get(
+      pluginId
+    )
+    console.log({ pluginSettings })
```
Comment (teleaziz) [inline plugins/localization/src/components/language-switcher.tsx:29]: Logs, in here and several other files

File: plugins/localization/src/components/language-switcher.tsx [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+      code: string
+      name: string
+    }[]
+  >()
+  const onLoad = async () => {
+    const currentUser = await context.user.getUser(context.user.id)
+    //@ts-ignore
+    const currentOrg = await context.user.organization
+    //@ts-ignore
+    const pluginSettings = await (await currentOrg.value.settings.plugins).get(
```
Comment (teleaziz) [inline plugins/localization/src/components/language-switcher.tsx:26]: Not a promise, 
```
settings = currentOrg.value.settings.plugins.get(pluginId)
```

**Review by teleaziz — CHANGES_REQUESTED** [review]
Thanks @jsjexpert  for the PR, left a few comments

**Comment by jsjexpert** [comment]
Great! Thanks for the review Aziz, will make some changes :)

File: plugins/localization/package.json (hunk 1 above)
Comment (jsjexpert) [reply plugins/localization/package.json]: Updated: @builder.io/plugin-localized-preview

Comment (jsjexpert) [reply plugins/localization/package.json]: updated: @builder.io/plugin-localized-preview

File: plugins/localization/src/components/language-switcher.tsx [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+const LangugeSwitcher = () => {
+  const [locales, setLocales] = useState<
+    {
+      code: string
+      name: string
+    }[]
+  >()
+  const onLoad = async () => {
+    const currentUser = await context.user.getUser(context.user.id)
```
[the remark this answers is not in the document]
Comment (jsjexpert) [reply plugins/localization/src/components/language-switcher.tsx]: updated. Context Typedefs appear incomplete though. Mentioned in source in latest commit. Search for @Aziz to find the comments :)

File: plugins/localization/src/components/language-switcher.tsx [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+    //@ts-ignore
+    const currentOrg = await context.user.organization
```
[the remark this answers is not in the document]
Comment (jsjexpert) [reply plugins/localization/src/components/language-switcher.tsx]: Updated

File: plugins/localization/src/components/language-switcher.tsx [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+          display: 'flex',
+          justifyContent: 'center',
+          alignItems: 'center',
+          cursor: 'pointer',
+          padding: '5px'
+        }}
+      >
+        <Gear />
+      </div>
+      <select
```
[the remark this answers is not in the document]
Comment (jsjexpert) [reply plugins/localization/src/components/language-switcher.tsx]: Updated. Much prettier now!

File: plugins/localization/src/models/localization.ts [hunk 10]
```diff
@@ -0,0 +1,5 @@
+export const localization = {
+  name: 'localization',
+  kind: 'data',
+  hideFromUI: true,
+}
```
[the remark this answers is not in the document]
Comment (jsjexpert) [reply plugins/localization/src/models/localization.ts:5]: File removed. Was not used

File: plugins/localization/src/plugin.ts (hunk 3 above)
Comment (jsjexpert) [reply plugins/localization/src/plugin.ts]: updated

File: plugins/localization/src/components/language-switcher.tsx (hunk 5 above)
Comment (jsjexpert) [reply plugins/localization/src/components/language-switcher.tsx]: removed logging except for caught errors

File: plugins/localization/src/components/language-switcher.tsx (hunk 6 above)
Comment (jsjexpert) [reply plugins/localization/src/components/language-switcher.tsx]: updated. Missing typedefs, managed via an extended interface but complete typedefs would be preferable

**Review by teleaziz — APPROVED** [review]

**Pull request merged**